### DISCUSSION TITLE
Number input field not accepting 0 and negative numbers with lazy rules.

### DISCUSSION THREAD

**Discussion by bala-techguy** [quasarframework/quasar#10074] [General - Components / Directives / etc] [answered]
I tried to create a new field of type "Number". Along with that I wanted to add a lazy rule for required criteria. 

By default 0 is not accepted as a value in the number field. 

Please check the below codepen. I want to have a validation rule where the field shouldn't be empty and at the same time it also accepts 0 as a value. 

https://codepen.io/bala41288/pen/KKmazQR

**Comment by bala-techguy** [comment, marked as answer]
Nevermind, I found a solution already. The below lazy rule worked for me. 

`:rules="[(val) => (val !== null && val !== '') || 'Field is required']"`

**Comment by pdanpdan** [comment]
You should better use isNaN
